earthvision/datasets/xview.py

```diff
@@ -1,2 +1,188 @@
+            chips: image file names
+            classes: classes for each ground truth
+        """
+        # check existnce
+        coords_path, coords_exists = self._check_exists_label('coords.npy')
+        chips_path, chips_exists = self._check_exists_label('chips.npy')
+        classes_path, classes_exists = self._check_exists_label('classes.npy')
+        
+        # if exist, load and return
+        if coords_exists and chips_exists and classes_exists:
+            coords = np.load(coords_path)
+            chips = np.load(chips_path)
+            classes = np.load(classes_path)
+            return coords, chips, classes
+        
+        # read xView_train.geojson
+        fname = os.path.join(self.root, 'xView_train.geojson')
+        with open(fname) as f:
+            data = json.load(f)
+
+        # initialize
+        coords = []
+        chips = []
+        classes = []
+
+        # extract
+        feat_len = len(data['features'])
+        img_files = os.listdir(os.path.join(self.root, self.resources[0].split('.')[0]))
+
+        for i in range(feat_len):
+            properties = data['features'][i]['properties']
+            b_id = properties['image_id']
+            val = [int(num) for num in properties['bounds_imcoords'].split(',')]
+
+            # type_id 75 and 82 don't belong to any class
+            # https://github.com/DIUx-xView/xView1_baseline/issues/3
+            if properties['type_id'] not in [75, 82] and b_id in img_files:
+                chips.append(b_id)
+                classes.append(properties['type_id'])
+                coords.append(val)
+        
+        # convert to numpy arrays and save
+        coords = np.array(coords)
+        chips = np.array(chips)
+        classes = np.array(classes)
+        np.save(coords_path, coords)
+        np.save(chips_path, chips)
+        np.save(classes_path, classes)
+        
+        return coords, chips, classes
+    
+    def __getitem__(self, idx):
+        """Returns a tensor image and a dictionary of target 
+        consists of bounding boxes and labels
+        """
+        if self.data_mode == 'train':
+            # image
+            img_path = os.path.join(self.root, 'train_images', self.chips[idx])
+            image = _load_img(img_path)
+            image = np.array(image)
+            image = torch.from_numpy(image)
+
+            # bounding box
+            bbox = self.coords[idx]
+            bbox = torch.from_numpy(bbox)
+
+            # label
+            label = self.classes[idx]
+            label = np.vectorize(index_mapping.get)(label)
+            label = torch.from_numpy(label)
+            
+            # combine bounding box and label
+            target = {}
+            target['boxes'] = bbox
+            target['labels'] = label
+
+            sample = (image, target)
+        elif self.data_mode == 'validation':
+            # image
+            img_path = os.path.join(self.root, 'val_images', self.chips[idx])
```

Comment:
Throw an error
```python
TypeError                                 Traceback (most recent call last)
<ipython-input-11-b4d4f6d7aaef> in <module>
----> 1 valid_data = next(iter(valid_dataloader))

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torch\utils\data\dataloader.py in __next__(self)
    433         if self._sampler_iter is None:
    434             self._reset()
--> 435         data = self._next_data()
    436         self._num_yielded += 1
    437         if self._dataset_kind == _DatasetKind.Iterable and \

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torch\utils\data\dataloader.py in _next_data(self)
    473     def _next_data(self):
    474         index = self._next_index()  # may raise StopIteration
--> 475         data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
    476         if self._pin_memory:
    477             data = _utils.pin_memory.pin_memory(data)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torch\utils\data\_utils\fetch.py in fetch(self, possibly_batched_index)
     42     def fetch(self, possibly_batched_index):
     43         if self.auto_collation:
---> 44             data = [self.dataset[idx] for idx in possibly_batched_index]
     45         else:
     46             data = self.dataset[possibly_batched_index]

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torch\utils\data\_utils\fetch.py in <listcomp>(.0)
     42     def fetch(self, possibly_batched_index):
     43         if self.auto_collation:
---> 44             data = [self.dataset[idx] for idx in possibly_batched_index]
     45         else:
     46             data = self.dataset[possibly_batched_index]

~\Documents\PythonNotebook\earth-vision\earthvision\datasets\xview.py in __getitem__(self, idx)
    168         elif self.data_mode == 'validation':
    169             # image
--> 170             img_path = os.path.join(self.root, 'val_images', self.chips[idx])
    171             image = _load_img(img_path)
    172             image = np.array(image)

TypeError: 'NoneType' object is not subscriptable
```

Reply:
Error raised since `self.chips` is None in 'validation' `data_mode`. Resolved by changing `self.chips` to `self.imgs`.